localstack/services/cloudformation/resource_provider.py

```diff
@@ -745,27 +519,12 @@ def execute_action(
             case _:
                 raise NotImplementedError(change_type)  # TODO: change error type
 
-    def should_use_legacy_provider(self, resource_type: str) -> bool:
-        # any config overwrites take precedence over the default order
-        return self.provider_config.get(resource_type) == "GenericBaseModel"
-
     def load_resource_provider(self, resource_type: str) -> ResourceProvider:
         # TODO: unify namespace of plugins
 
-        # opt-in to switch to older resource provider
-        if self.should_use_legacy_provider(resource_type):
-            return self._load_legacy_resource_provider(resource_type)
-
         # 1. try to load pro resource provider
         # prioritise pro resource providers
         if PRO_RESOURCE_PROVIDERS:
-            # temporary patch until this has equivalent resource providers in -ext
-            if resource_type in {
-                "AWS::ECR::Repository",
-                "AWS::SecretsManager::SecretTargetAttachment",
-                "AWS::EC2::SubnetRouteTableAssociation",
```

Comment:
this especially might cause some new issues with -ext since we didn't test them yet because of this exemption list!